PowerDNS 4.9.2 fails to build with gcc15

- Program: Authoritative
 - Issue type: Bug report

### Short description
PowerDNS 4.9.2 fails to build to build with gcc15

https://koji.fedoraproject.org/koji/taskinfo?taskID=128026942
Buildlog: https://kojipkgs.fedoraproject.org//work/tasks/7058/128027058/build.log

make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
In file included from webserver.hh:34,
                 from ixfrdist-web.hh:23,
                 from ixfrdist-web.cc:22:
credentials.hh:50:55: error: ‘uint64_t’ has not been declared
   50 | std::string hashPassword(const std::string& password, uint64_t workFactor, uint64_t parallelFactor, uint64_t blockSize);
      |                                                       ^~~~~~~~
credentials.hh:1:1: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
  +++ |+#include <cstdint>
    1 | /*
credentials.hh:50:76: error: ‘uint64_t’ has not been declared
   50 | std::string hashPassword(const std::string& password, uint64_t workFactor, uint64_t parallelFactor, uint64_t blockSize);
      |                                                                            ^~~~~~~~
credentials.hh:50:76: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
credentials.hh:50:101: error: ‘uint64_t’ has not been declared
   50 | std::string hashPassword(const std::string& password, uint64_t workFactor, uint64_t parallelFactor, uint64_t blockSize);
      |                                                                                                     ^~~~~~~~
credentials.hh:50:101: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
credentials.hh:52:77: error: ‘uint64_t’ has not been declared
   52 | bool verifyPassword(const std::string& binaryHash, const std::string& salt, uint64_t workFactor, uint64_t parallelFactor, uint64_t blockSize, const std::string& binaryPassword);
      |                                                                             ^~~~~~~~
credentials.hh:52:77: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
credentials.hh:52:98: error: ‘uint64_t’ has not been declared
   52 | bool verifyPassword(const std::string& binaryHash, const std::string& salt, uint64_t workFactor, uint64_t parallelFactor, uint64_t blockSize, const std::string& binaryPassword);
      |                                                                                                  ^~~~~~~~
credentials.hh:52:98: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
credentials.hh:52:123: error: ‘uint64_t’ has not been declared
   52 | bool verifyPassword(const std::string& binaryHash, const std::string& salt, uint64_t workFactor, uint64_t parallelFactor, uint64_t blockSize, const std::string& binaryPassword);
      |                                                                                                                           ^~~~~~~~
credentials.hh:52:123: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
credentials.hh:84:10: error: ‘uint64_t’ does not name a type
   84 |   static uint64_t const s_defaultWorkFactor;
      |          ^~~~~~~~
credentials.hh:84:10: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
credentials.hh:85:10: error: ‘uint64_t’ does not name a type
   85 |   static uint64_t const s_defaultParallelFactor;
      |          ^~~~~~~~
credentials.hh:85:10: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
credentials.hh:86:10: error: ‘uint64_t’ does not name a type
   86 |   static uint64_t const s_defaultBlockSize;
      |          ^~~~~~~~
credentials.hh:86:10: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
credentials.hh:93:3: error: ‘uint64_t’ does not name a type
   93 |   uint64_t d_workFactor{0};
      |   ^~~~~~~~
credentials.hh:93:3: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
credentials.hh:94:3: error: ‘uint64_t’ does not name a type
   94 |   uint64_t d_parallelFactor{0};
      |   ^~~~~~~~
credentials.hh:94:3: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
credentials.hh:95:3: error: ‘uint64_t’ does not name a type
   95 |   uint64_t d_blockSize{0};
      |   ^~~~~~~~
credentials.hh:95:3: note: ‘uint64_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
credentials.hh:97:3: error: ‘uint32_t’ does not name a type
   97 |   uint32_t d_fallbackHashPerturb{0};
      |   ^~~~~~~~
credentials.hh:97:3: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
credentials.hh:98:3: error: ‘uint32_t’ does not name a type
   98 |   uint32_t d_fallbackHash{0};
      |   ^~~~~~~~
credentials.hh:98:3: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
make[3]: *** [Makefile:3382: ixfrdist-web.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1  -I/usr/include/luajit-2.1       -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o sdig-rcpgenerator.o `test -f 'rcpgenerator.cc' || echo './'`rcpgenerator.cc
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1  -I/usr/include/luajit-2.1    -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o ixfrdist-stats.o ixfrdist-stats.cc
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1  -I/usr/include/luajit-2.1       -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o sdig-statbag.o `test -f 'statbag.cc' || echo './'`statbag.cc
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1  -I/usr/include/luajit-2.1    -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o epollmplexer.o epollmplexer.cc
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1  -I/usr/include/luajit-2.1    -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o calidns.o calidns.cc
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1  -I/usr/include/luajit-2.1    -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o webserver.o webserver.cc
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1  -I/usr/include/luajit-2.1    -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o zone2ldap.o zone2ldap.cc
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1  -I/usr/include/luajit-2.1       -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o sdig-dnsrecords.o `test -f 'dnsrecords.cc' || echo './'`dnsrecords.cc
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1  -I/usr/include/luajit-2.1    -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o dnstcpbench.o dnstcpbench.cc
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1  -I/usr/include/luajit-2.1    -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o dnsbulktest.o dnsbulktest.cc
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1  -I/usr/include/luajit-2.1    -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o ixfrdist.o ixfrdist.cc
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.9.2-build/pdns-4.9.2/pdns'
In file included from resolver.hh:41,
                 from axfr-retriever.hh:27,
                 from ixfrdist.cc:44:
dnssecinfra.hh: In member function ‘void DNSCryptoKeyEngine::createFromPEMString(DNSKEYRecordContent&, const std::string&)’:
dnssecinfra.hh:70:51: warning: ignoring attributes on template argument ‘int (*)(FILE*)’ [-Wignored-attributes]
   70 |       unique_ptr<std::FILE, decltype(&std::fclose)> inputFile{fmemopen(const_cast<char*>(contents.data()), contents.length(), "r"), &std::fclose};
      |                                                   ^
dnssecinfra.hh: In member function ‘std::string DNSCryptoKeyEngine::convertToPEMString() const’:
dnssecinfra.hh:93:51: warning: ignoring attributes on template argument ‘int (*)(FILE*)’ [-Wignored-attributes]
   93 |       unique_ptr<std::FILE, decltype(&std::fclose)> outputFile{fmemopen(output.data(), output.length() - 1, "w"), &std::fclose};
      |                                                   ^
make[2]: *** [Makefile:2834: all] Error 2
make[1]: *** [Makefile:605: all-recursive] Error 1
make: *** [Makefile:537: all] Error 2

--- a/pdns/credentials.hh
+++ b/pdns/credentials.hh
@@ -21,7 +21,7 @@
  */
 #pragma once
 
-#include <memory>
+#include <cstdint>
 #include <string>
 
 class SensitiveData

